### PR TITLE
feat: support loading of new files automatically with `--live`

### PR DESCRIPTION
**Pull request by imnotjames**
Implements a "watcher" on files being added and removed when `--live` is specified in the command line arguments.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by imnotjames** [comment]
This is a second attempt at #53 now that I understand the problem a bit better :)

File: src/runtime/server.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ import express, {
 } from 'express';
 import userAgentMiddleware from 'express-useragent';
 import nocache from 'nocache';
+import {printRouteInfo} from '../printers/start';
+import watch from 'node-watch';
```
Comment (dkundel) [inline src/runtime/server.ts]: Could you try this library instead? https://www.npmjs.com/package/chokidar? It seems like this is a defacto standard. It's being used by libraries like nodemon, babel and browser-sync

File: src/runtime/server.ts [hunk 2]
```diff
@@ -77,8 +80,44 @@ export async function createServer(
...
+  let routes = await getFunctionsAndAssets(config.baseDir);
+  let routeMap = setRoutes(routes);
+
+  if (config.live) {
+    let seenFiles = new Set();
+
+    watch(config.baseDir, { recursive: true }, async (evt, name) => {
+      let relativeName = path.relative(config.baseDir, name);
+
+      if (!relativeName.match(/^(assets|functions)\//)) {
```
Comment (dkundel) [inline src/runtime/server.ts]: If you could check out the library mentioned above, it seems like it supports creating a glob pattern. So we could be even more precise. Something like:
```
chokidar.watch(`${config.baseDir}/{functions/**/*.js,assets/**/*}`)
  .on('add', ...)
  .on('remove', ...)
```

File: src/runtime/server.ts (hunk 1 above)
Comment (imnotjames) [reply src/runtime/server.ts]: That's a much better library to use for this use case - I didn't realize that babel was using that!  I'll definitely make that change.

File: src/runtime/server.ts [hunk 3]
```diff
@@ -77,8 +80,44 @@ export async function createServer(
...
+        // Only reload on assets and functions.
+        // No other files are very important watching.
+        return;
+      }
+
+      if (evt == 'update') {
+        if (name in seenFiles) {
```
[the remark this answers is not in the document]
Comment (imnotjames) [reply src/runtime/server.ts]: Odd - I'd investigate it but given that I'm switching this to chokidar we should be fine.

File: src/runtime/server.ts [hunk 4]
```diff
@@ -77,8 +80,44 @@ export async function createServer(
...
+    watch(config.baseDir, { recursive: true }, async (evt, name) => {
+      let relativeName = path.relative(config.baseDir, name);
+
+      if (!relativeName.match(/^(assets|functions)\//)) {
+        // Only reload on assets and functions.
+        // No other files are very important watching.
+        return;
+      }
+
+      if (evt == 'update') {
```
[the remark this answers is not in the document]
Comment (imnotjames) [reply src/runtime/server.ts]: No longer relevant by switching to chokidar.

File: src/runtime/server.ts (hunk 2 above)
Comment (imnotjames) [reply src/runtime/server.ts]: I've done this.  I used instead:

```
    const watcher = chokidar.watch(
      [
        path.join(config.baseDir, '/functions/**/*.js'),
        path.join(config.baseDir, '/assets/'),
      ],
...
    )
```

**Pull request merged**